From c9c11e688c9e693c9a0fed057f5498de71d3d98b Mon Sep 17 00:00:00 2001
From: yaozenhu <yaozenhu@realtek.com>
Date: Wed, 9 Nov 2016 11:23:14 +0800
Subject: [PATCH 4/7] [fix bug] send flush event to baseparse plugin cause
 deadlock [solution] baseparse pause task when receiving flush start event,
 start task until receving flush stop event [reviewer] Paul

Change-Id: If00004de15cad6dbb089875496d612b1b8e3aa01
---
 libs/gst/base/gstbaseparse.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libs/gst/base/gstbaseparse.c b/libs/gst/base/gstbaseparse.c
index 45bc869..c4f9271 100644
--- a/libs/gst/base/gstbaseparse.c
+++ b/libs/gst/base/gstbaseparse.c
@@ -1331,6 +1331,7 @@ gst_base_parse_sink_event_default (GstBaseParse * parse, GstEvent * event)
       break;
 
     case GST_EVENT_FLUSH_START:
+      gst_pad_pause_task (parse->sinkpad);
       GST_OBJECT_LOCK (parse);
       parse->priv->flushing = TRUE;
       GST_OBJECT_UNLOCK (parse);
@@ -1347,6 +1348,8 @@ gst_base_parse_sink_event_default (GstBaseParse * parse, GstEvent * event)
       parse->priv->skip = 0;
 
       forward_immediate = TRUE;
+      gst_pad_start_task (parse->sinkpad,
+        (GstTaskFunction) gst_base_parse_loop, parse->sinkpad, NULL);
       break;
 
     case GST_EVENT_EOS:
-- 
2.7.4

